org.sonarlint.eclipse.ui/src/org/sonarlint/eclipse/ui/internal/command/MarkAsResolvedCommand.java

```diff
@@ -55,10 +64,51 @@ private static Optional<ResolvedBinding> getBinding(IMarker marker) {
   }
 
   @Override
-  protected void execute(IMarker selectedMarker) {
-    var msg = new MessageBox(Display.getDefault().getActiveShell());
-    msg.setText("TODO");
-    msg.open();
+  protected void execute(IMarker selectedMarker, IWorkbenchWindow window) {
+    var checkJob = new Job("Check user permissions for administering issues") {
+
+      private CheckStatusChangePermittedResponse result;
+      private ResolvedBinding resolvedBinding;
+
+      @Override
+      protected IStatus run(IProgressMonitor monitor) {
+        var issueKey = selectedMarker.getAttribute(MarkerUtils.SONAR_MARKER_SERVER_ISSUE_KEY_ATTR, null);
+        if (issueKey == null) {
+          return Status.error("No issue key for this marker");
+        }
+        var binding = getBinding(selectedMarker);
+        if (binding.isPresent()) {
+          resolvedBinding = binding.get();
```

Comment:
This class variable can be removed entirely, or am I mistaken?

Reply:
It is passed to the downstream method after the job has completed